File: 2025/day01/one.py
import sys
from os.path import dirname, join


FILE = join(dirname(__file__), '../../data/2025/day01/', sys.argv[-1])
with open(FILE) as f:
    data = f.readlines()


def naive(shifts):
    current = 50
    passes = 0
    for shift in shifts:
        current += int(shift.replace('L', '-').replace('R', '+'))
        current = current % 100
        
        if current == 0:
            passes += 1

    return passes


print(naive(data))

# from timeit import repeat
# print(repeat(
#       'naive(data)',
#       setup='from __main__ import naive',
#       globals=globals(),
#       repeat=3,
#       number=10,
# ))

File: 2025/day01/one.rb
data = File.readlines(
  File.expand_path("../../../data/2025/day01/#{ARGV[-1]}", __FILE__),
)


def count_zeros(instructions)
  position = 50

  instructions
    .each{ |s| s.sub!(/[LR]/, 'L' => '-', 'R' => '+') }
    .map!( &:to_i )
    .map!{ |i| (position += i) }
    .count{ |i| i % 100 == 0 }

end


puts(count_zeros(data))
